From 48cac2f53c5fdafbd8e1caf6911f68a10f54398d Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Mon, 7 May 2018 14:07:17 +0200
Subject: [PATCH 1/9] RIL: Allow overriding RadioResponse and RadioIndication

* Create RadioResponse and RadioIndication with helper methods
* Make the constructor for RadioIndication public

Change-Id: I57990fb7a25c6b26f633a0f660c5d355c53bc9bb
---
 src/java/com/android/internal/telephony/RIL.java     | 12 ++++++++++--
 .../android/internal/telephony/RadioIndication.java  |  2 +-
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/java/com/android/internal/telephony/RIL.java b/src/java/com/android/internal/telephony/RIL.java
index fac9266d5..6c412f13c 100644
--- a/src/java/com/android/internal/telephony/RIL.java
+++ b/src/java/com/android/internal/telephony/RIL.java
@@ -464,8 +464,8 @@ public class RIL extends BaseCommands implements CommandsInterface {
                 Context.CONNECTIVITY_SERVICE);
         mIsMobileNetworkSupported = cm.isNetworkSupported(ConnectivityManager.TYPE_MOBILE);
 
-        mRadioResponse = new RadioResponse(this);
-        mRadioIndication = new RadioIndication(this);
+        mRadioResponse = createRadioResponse(this);
+        mRadioIndication = createRadioIndication(this);
         mOemHookResponse = new OemHookResponse(this);
         mOemHookIndication = new OemHookIndication(this);
         mRilHandler = new RilHandler();
@@ -493,6 +493,14 @@ public class RIL extends BaseCommands implements CommandsInterface {
         getOemHookProxy(null);
     }
 
+    protected RadioResponse createRadioResponse(RIL ril) {
+        return new RadioResponse(ril);
+    }
+
+    protected RadioIndication createRadioIndication(RIL ril) {
+        return new RadioIndication(ril);
+    }
+
     @Override
     public void setOnNITZTime(Handler h, int what, Object obj) {
         super.setOnNITZTime(h, what, obj);
diff --git a/src/java/com/android/internal/telephony/RadioIndication.java b/src/java/com/android/internal/telephony/RadioIndication.java
index f7a7943e4..b3fd47216 100644
--- a/src/java/com/android/internal/telephony/RadioIndication.java
+++ b/src/java/com/android/internal/telephony/RadioIndication.java
@@ -108,7 +108,7 @@ import java.util.List;
 public class RadioIndication extends IRadioIndication.Stub {
     RIL mRil;
 
-    RadioIndication(RIL ril) {
+    public RadioIndication(RIL ril) {
         mRil = ril;
     }
 
-- 
2.17.1

